hoomd/hpmc/update.py

```diff
@@ -398,114 +398,145 @@ def enable(self):
+                                                   be updated.
+        py_updater (`callable`): the python callback that performs the update
+                                 moves. This must be a python method that is a
+                                 function of the timestep of the simulation.
+               It must actually update the :py:class:`hoomd.hpmc.field.wall`)
+               managed object.
+        move_ratio (float): the probability with which an update move is
+                            attempted.
+        seed (int): the seed of the pseudo-random number generator that
+                    determines whether or not an update move is attempted
+        trigger (int): the number of timesteps between update move attempt
+                       attempts.
+
+    Every *trigger* steps, a walls update move is tried with probability
+    *move_ratio*. This update move is provided by the *py_updater* callback.
+    Then, update.wall only accepts an update move provided by the python
+    callback if it maintains confinement conditions associated with all walls.
+    Otherwise, it reverts back to anon-updated copy of the walls.
+
+    Once initialized, the update provides the following log quantities that can
+    be logged via :py:class:`hoomd.analyze.log`:
+
+    * **hpmc_wall_acceptance_ratio** - the acceptance ratio for wall update
+    moves
 
-    Once initialized, the update provides the following log quantities that can be logged via :py:class:`hoomd.analyze.log`:
+    Examples::
 
-    * **hpmc_wall_acceptance_ratio** - the acceptance ratio for wall update moves
+        TODO: link to example notebook, move examples here to notebook
 
     Example::
 
         mc = hpmc.integrate.sphere(seed = 415236);
         ext_wall = hpmc.compute.wall(mc);
-        ext_wall.add_sphere_wall(radius = 1.0, origin = [0, 0, 0], inside = True);
+        ext_wall.add_sphere_wall(radius = 1.0, origin = [0, 0, 0],
+                                 inside = True);
         def perturb(timestep):
           r = np.sqrt(ext_wall.get_sphere_wall_param(index = 0, param = "rsq"));
-          ext_wall.set_sphere_wall(index = 0, radius = 1.5*r, origin = [0, 0, 0], inside = True);
-        wall_updater = hpmc.update.wall(mc, ext_wall, perturb, move_ratio = 0.5, seed = 27, period = 50);
-        log = analyze.log(quantities=['hpmc_wall_acceptance_ratio'], period=100, filename='log.dat', overwrite=True);
+          ext_wall.set_sphere_wall(index = 0, radius = 1.5*r,
+                                   origin = [0, 0, 0], inside = True);
+        wall_updater = hpmc.update.wall(mc, ext_wall, perturb,
+                                        move_ratio = 0.5, seed = 27,
+                                        trigger = 50);
+        log = analyze.log(quantities=['hpmc_wall_acceptance_ratio'],
+                          trigger=100, filename='log.dat', overwrite=True);
 
     Example::
 
         mc = hpmc.integrate.sphere(seed = 415236);
```

Comment:
At some point we will have to make this have a deferred initialization for the walls. Since we want to be able to create walls without having a mc integrator instance at the time of creation. They need the same attach concept as well.

Reply:
Opened new issue since I think it's outside the scope for these specific changes. https://github.com/glotzerlab/hoomd-blue/issues/551